### PR TITLE
POC: Slippi visualiser on stats

### PR DESCRIPTION
**Pull request by Fcornaire**
# This is a Proof of concept and not merge ready !

First time contribution here, i added the visualiser used in the preview on slippi.gg/tournament in the stats page.
visualiser

It only appear with large combo/conversion (Red heart) and play only on focus.
This PR is not merge ready. I want feedback and discussion about my implementation since there is still a lot other thing to fix mostly in the [react wrapper](https://github.com/Fcornaire/react-slippi-visualiser) that i made.

Basically, i made a wrapper that take a replay (parser) and send it to the slippi-visualiser original lib throught react.
The desktop app only need a tweak about Jquery and to load all the animations in the html.
(i also had to update react since i use hook in the wrapper 😅)

What's do you think about this ?

### CODE REVIEW

**Comment by JLaferri** [comment]
I think it's a cool idea but right now it's way too big.

If anything maybe it should be behind a click and open in a dialog or something. But also we're currently working on a refactor of the desktop app so im not sure whether something like this would get merged.

Maybe @vinceau has thoughts.

**Comment by Fcornaire** [comment]
oh yeah ofc, not a problem if it's not implemented right now or never, was just testing to see how it will turn out.
I also heard about the rewrite but i thought it's being made internaly ?

like a said it's just a POC, i thought about the dialog but my problem with it was like you have and extra step to show something that should be quick to see, since the visualiser is not a replacement of the replay itself.
i think it's cool seeing quick highlights but obviously it's not final or anything, so 😅

**Comment by vinceau** [comment]
@Fcornaire This is a very cool concept! Nice work! There's definitely benefit to having the game show up in the desktop app and without having to load it through Dolphin. I agree with @JLaferri that it probably should be hidden behind a click in a dialog, more like how slippi.gg shows the game previews. We should also clearly distinguish between playing it back in browser verus playing it back in Dolphin (added in #105).

Some comments regarding the code and implementation:

1. **The Rewrite**. For starters, I don't think we should be accepting anymore PRs into the "legacy" desktop app. As per #112, we'll be rewriting the desktop app in Typescript so new features should have PRs pointed to the [`refactor/rewrite`](https://github.com/project-slippi/slippi-desktop-app/tree/refactor/rewrite). Ideally imported libraries are also written in Typescript or provide `.d.ts` typings out of the box.
2. **Global imports**. I'm not a big fan of having to globally import jQuery and all the animations required for getting the game preview to work. Since you've added them to the root `app.html` they will always be loaded regardless of page, and regardless if the game is shown. Ideally we should not have to do any global imports (including global CSS) to use the `react-slippi-visualiser` component.
3. **Depending on jQuery is not great...**. React is already a fairly big library. jQuery is an even bigger, chonkier boi. So use of jQuery should be avoided. Additionally, since jQuery modifies the DOM outside of React's shadow-DOM we lose out on the benefits of using React in the first place. Ideally we should not have to depend on jQuery at all and instead rely on passing React refs for DOM manipulation.
4. **Melee light/visualizer is still largely incomplete**. Out of the 26 character cast, only 5 characters are supported (<20%). By including all the animations (~30MB) in a [compiled and minified format](https://github.com/Fcornaire/react-slippi-visualiser/blob/master/public/animations.js) it makes expanding the number of supported characters very difficult. Ideally the animations are included in the React library itself and can be individually lazy-loaded when needed.

I've messed around a bit with [react-slp-viewer](https://github.com/vinceau/react-slp-viewer) to try and fix some of these issues. It supports Typescript, does not depend on global imports, and lazy-loads the animations on demand. Let me know what you think about these comments and ideas, and if you would like to collaborate on this together.

**Comment by Fcornaire** [comment]
@vinceau & @JLaferri  Thanks for the feedback. 
You mostly answered the thing i thought needed some rework:
 1. I'm gonna wait for now since this branch [refactor/rewrite](https://github.com/project-slippi/slippi-desktop-app/tree/refactor/rewrite) probably gonna change a lot, so i will likely wait for the full re write since this feature is not mandatory (i could also help on the refacto but i feel like my help is not really needed 😅 )
 2. Yup , didn't  liked to do that but it's was mandatory for the visualiser to work. Again it's just a POC using what is already there to see how thing will work
 3. the ideal will be no JQuery, the original lib need rework
 4. I actually searched the source but only found the compiled version 😥. Was not a fan of loading all that in the html  but it was the only solution i found for now at least. I think ideally the original lib should manage that
 
I'm gonna look into your implementation  [react-slp-viewer](https://github.com/vinceau/react-slp-viewer) and try to go from there.

I'm also closing this PR since it's useless now
